R/zchunk_LB125.LC_tot.R

```diff
@@ -39,65 +42,128 @@ module_aglu_LB125.LC_tot_DISABLED <- function(command, ...) {
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # ===================================================
+
+    # -----------------------------------------------------------------------------
+    # Perform computations
+    # Create a table with all land types
+    bind_rows(L123.LC_bm2_R_MgdPast_Yh_GLU,
+              L123.LC_bm2_R_MgdFor_Yh_GLU,
+              L124.LC_bm2_R_Shrub_Yh_GLU_adj,
+              L124.LC_bm2_R_Grass_Yh_GLU_adj,
+              L124.LC_bm2_R_UnMgdPast_Yh_GLU_adj,
+              L124.LC_bm2_R_UnMgdFor_Yh_GLU_adj) %>%
+      # The next two lines are TEMPORARY - because currently we're using temp-data-inject data
+      # When using final data, don't need to reshape them
+      gather(year, value, -GCAM_region_ID, -Land_Type, -GLU) %>%
+      mutate(year = as.integer(substr(year, 2, 5))) %>%
+      bind_rows(L120.LC_bm2_R_UrbanLand_Yh_GLU,
+                L120.LC_bm2_R_Tundra_Yh_GLU,
+                L120.LC_bm2_R_RckIceDsrt_Yh_GLU,
+                L122.LC_bm2_R_HarvCropLand_Yh_GLU,
+                L122.LC_bm2_R_OtherArableLand_Yh_GLU)  ->
+      L125.LC_bm2_R_LT_Yh_GLU
+
+    # Adding up total land area by region, GLU, and year
+    L125.LC_bm2_R_LT_Yh_GLU %>%
+      group_by(GCAM_region_ID, GLU, year) %>%               # group the data by GCAM_region_id and GLU
+      summarise(value = sum(value)) ->                      # Adding up total land area by region, year, and GLU
+      L125.LC_bm2_R_Yh_GLU
+
+    # It's necessary to make sure the Land Cover changing rates are in certain tolerances
+    L125.LC_bm2_R_Yh_GLU %>%
+      arrange(GCAM_region_ID, GLU, year) %>%
+      mutate(changing_rate = value / lag(value)) %>%          # calculate the changing rate
+      replace_na(list(changing_rate = 1)) %>%                 # assign na to 1 (no change)
+      ungroup() %>%
+      select(changing_rate) ->
+      LC_check
+
+    # Stop if the changing rate out of the tolerance boundaries
+    if (max(abs(LC_check - 1)) > LAND_TOLERANCE)
+    {stop("ERROR: Interannual fluctuation in global land cover exceeds tolerance threshold")}
+
+    #Write out the totals, by region and by region x GLU
+    L125.LC_bm2_R_Yh_GLU %>%
+      filter(year == min(year)) %>%                        # using the starting year only
+      group_by(GCAM_region_ID, GLU) %>%                    # group by GCAM_region_ID and GLU
+      summarise(value = sum(value)) %>%                    # summarize area by region and GLU
+      rename(LC_bm2 = value) ->                            # rename the column
```

Comment:
@bpbond I've revised it according to your comments, but syncing is failed. Do you have a sense of why?